community/modules/scheduler/k8s-cluster/main.tf

```diff
@@ -0,0 +1,261 @@
+/**
+  * Copyright 2022 Google LLC
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+locals {
+  dash             = var.prefix_with_deployment_name && var.name_suffix != "" ? "-" : ""
+  prefix           = var.prefix_with_deployment_name ? var.deployment_name : ""
+  name_maybe_empty = "${local.prefix}${local.dash}${var.name_suffix}"
+  name             = local.name_maybe_empty != "" ? local.name_maybe_empty : "NO-NAME-GIVEN"
+
+  cluster_authenticator_security_group = var.authenticator_security_group == null ? [] : [{
+    security_group = var.authenticator_security_group
+  }]
+
+  sa_email = var.service_account.email != null ? var.service_account.email : data.google_compute_default_service_account.default-sa.email
+}
+
+data "google_compute_default_service_account" "default-sa" {
+  project = var.project_id
+}
+
+resource "google_container_cluster" "gke-cluster" {
+  provider = google-beta
+
+  project         = var.project_id
+  name            = local.name
+  location        = var.region
+  resource_labels = var.labels
+
+  # decouple node pool lifecyle from cluster life cycle
+  remove_default_node_pool = true
+  initial_node_count       = 1 # must be set when remove_default_node_pool is set
+
+  network    = var.network_id
+  subnetwork = var.subnetwork_self_link
+
+  # Note: the existence of the "master_authorized_networks_config" block enables
+  # the master authorized networks even if it's empty.
+  master_authorized_networks_config {
+  }
+
+  private_ipv6_google_access = var.enable_private_ipv6_google_access ? "PRIVATE_IPV6_GOOGLE_ACCESS_TO_GOOGLE" : null
+
+  master_auth {
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+
+  pod_security_policy_config {
+    enabled = true
+  }
+
+  enable_shielded_nodes = true
```

Comment:
Does this makes all pods as shielded nodes? or does this just allow for shielded nodes to exist?

Reply:
no, just bringing awareness. Ok to leave it as is.